Mage.Tests/src/test/java/org/mage/test/cards/cost/adventure/CastAdventureCardsTest.java

```diff
@@ -0,0 +1,50 @@
+package org.mage.test.cards.cost.adventure;
+
+import mage.constants.PhaseStep;
+import mage.constants.Zone;
+import org.junit.Test;
+import org.mage.test.serverside.base.CardTestPlayerBase;
+
+public class CastAdventureCardsTest extends CardTestPlayerBase {
+    @Test
+    public void testCastTreatsToShare() {
+        addCard(Zone.BATTLEFIELD, playerA, "Forest");
+        addCard(Zone.HAND, playerA, "Curious Pair");
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, "Treats to Share");
+        setStopAt(1, PhaseStep.BEGIN_COMBAT);
+        execute();
+        assertPermanentCount(playerA, "Food", 1);
+        assertExileCount(playerA, "Curious Pair", 1);
+        assertGraveyardCount(playerA,0);
+    }
+
+    @Test
+    public void testCastCuriousPair() {
+        addCard(Zone.BATTLEFIELD, playerA, "Forest");
+        addCard(Zone.BATTLEFIELD, playerA, "Forest");
+        addCard(Zone.HAND, playerA, "Curious Pair");
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, "Curious Pair");
+        setStopAt(1, PhaseStep.BEGIN_COMBAT);
+        execute();
+        assertPermanentCount(playerA, "Food", 0);
```

Comment:
Use that commands instead simple execute:
```
setStrictChooseMode(true); // raise error on missing choose command in test
setStopAt(1, PhaseStep.END_TURN);
execute();
assertAllCommandsUsed(); // raise error on wrong (not used) command in test
``

Reply:
Great, thanks